modules/configloader/src/BoardConf.cxx

```diff
@@ -0,0 +1,147 @@
+/*
+ *  BoardConf.cxx
+ *  Copyright 2020 ItJustWorksTM
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+#include <concepts>
+#include <fstream>
+#include <optional>
+#include <nameof.hpp>
+#include <range/v3/algorithm/transform.hpp>
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wambiguous-reversed-operator"
+#include <rapidjson/document.h>
+#include <rapidjson/istreamwrapper.h>
+#include <rapidjson/schema.h>
+#pragma clang diagnostic pop
+
+#include "BoardConf.hxx"
+
+template <class T, class C>
+struct MptrKeyPair {
+    using FieldType = T;
+    using ClassType = C;
+
+    T C::* mptr;
+    const char* key;
+
+    constexpr MptrKeyPair(T C::* mptr, const char* key) noexcept : mptr{mptr}, key{key} {}
+};
+
+template <class T>
+concept MpKPair = std::is_same_v<T, MptrKeyPair<typename T::FieldType, typename  T::ClassType>>;
+
+#define ADDR_AND_NAME(var) MptrKeyPair{&var, NAMEOF(var).c_str()}
```

Comment:
Macros are a codesmell, sure, but here we use it for reflection, and C++ currently has no better way of doing that kind of static reflection.
And since we're already using macros, we might as well avoid the duplication of the variable name, which does avoid having a mismatched key and member function pointer

The `.c_str()` on a pr-value does look odd too, but the way the macro is used benefits from lifetime-extension rules, so while it is a footgun, it is fine to use if wielded as shown below.